src/app/static/src/tests/e2e/review-input.spec.ts

```diff
@@ -10,12 +11,137 @@ test('can view review input plots', async ({ projectPage }) => {
 
     // Continue button is active
     const continueButton = page.locator("#continue").nth(0);
-    await expect(continueButton).not.toHaveClass("disabled");
+    await expect(continueButton).not.toBeDisabled();
 
     // When I click continue
     await continueButton.click();
 
     // Review inputs page matches screenshot after data fetched
-    await expect(page.locator("#review-loading")).not.toBeVisible();
-    await expect(page.locator("#review-inputs")).toHaveScreenshot("review-input-landing.png");
+    await expect(page.locator("#review-loading")).toHaveCount(0, {timeout: 10000});
+    await expect(page.locator("#review-inputs")).toHaveScreenshot("time-series-landing.png");
+    const waitForPlotUpdate = await buildWaitForUpdateCallback(page.locator(".plot-container.plotly"), 100, 20);
```

Comment:
can probably also do below but i think your method will be faster since youre just checking for the innerHTML, i would still vote for the way youve done it but just in case you wanted a more playwright-y alternative
```ts
await expect.poll(() => locator.screenshot()).toMatchScreenshot([SS])
```

Reply:
Nice! So I couldn't get this working but I did get it to work using expect.toPass https://playwright.dev/docs/test-assertions#expecttopass which I think addresses what we need. That looks good to me, and means we don't have to do this plot update every time we change something. Can just wrap the offending slow updating requests in an expect.toPass.